tabbycat/results/result.py

```diff
@@ -693,202 +589,376 @@ def as_dicts(self):
+    def get_ranked_dt(self):
+        return [self.debateteams[s] for s in self.scoresheet.ranked_sides()]
 
-    def teamscorefield_points(self, side):
-        return 4 - self.scoresheet.rank(side)
+    # --------------------------------------------------------------------------
+    # Management methods
+    # --------------------------------------------------------------------------
 
-    def teamscorefield_score(self, side):
-        return self.scoresheet.get_total(side)
+    def identical(self, other):
+        return super().identical(other) and self.scoresheet.identical(other.scoresheet)
 
+    # --------------------------------------------------------------------------
+    # Team score fields
+    # --------------------------------------------------------------------------
 
-class BPEliminationDebateResult(BaseDebateResult):
-    """For British Parliamentary elimination rounds.  Does not take speaker
-    identities, speaker scores or ranks.  Instead, it just notes the two
-    advancing teams, using the `win` field, which is set to True for both of
-    the advancing teams."""
+    def teamscore_field_points(self, side):
+        if not hasattr(self.scoresheet, 'rank'):
+            return None
+        return len(self.sides) - self.scoresheet.rank(side)
 
-    is_voting = False
-    uses_advancing = True
+    def teamscore_field_win(self, side):
+        return side in self.scoresheet.winners()
+
+    def as_dicts(self):
+        yield {'teams': self.sheet_as_dicts(self.scoresheet)}
+
+
+class ConsensusDebateResultWithScores(DebateResultWithScoresMixin, ConsensusDebateResult):
+    """Consensus debate result including speaker scores"""
+
+    def get_scoresheet_class(self):
+        if len(self.sides) == 2:
+            return super().get_scoresheet_class()
+        elif len(self.sides) == 4:
+            return BPScoresheet
+
+    def load_scoresheets(self):
+        super().load_scoresheets()
+
+        speakerscore = self.ballotsub.speakerscore_set.filter(
+            debate_team__side__in=self.sides,
+            position__in=self.positions,
+        ).select_related('debate_team')
+
+        for ss in speakerscore:
+            self.set_score(ss.debate_team.side, ss.position, ss.score)
+
+    def set_score(self, side, position, score):
+        self.scoresheet.set_score(side, position, score)
 
     # --------------------------------------------------------------------------
-    # Management methods
+    # Model fields
     # --------------------------------------------------------------------------
 
-    def init_blank_buffer(self):
-        super().init_blank_buffer()
-        # To be complete, this list must have two sides in it, but we don't
-        # enforce anything before checking completeness.
-        self.advancing = []
+    def speakerscore_field_score(self, side, position):
+        return self.scoresheet.get_score(side, position)
+    get_score = speakerscore_field_score
```

Comment:
Very minor but maybe put an empty line above this line? Without the blank link it looks like a typo lol
(If flake8 complains about it we should reconfigure flake8)
(I think I find this a little different to similar patterns in `admin.py` files because this line defines a new function, as opposed to modifies an existing one. But as it is is fine if you would find a blank line more confusing.)